Userland/Libraries/LibC/malloc.cpp

```diff
@@ -455,6 +455,20 @@ void* realloc(void* ptr, size_t size)
     return new_ptr;
 }
 
+void* reallocarray(void *ptr, size_t nmemb, size_t size)
+{
+	if (!nmemb)
+		return nullptr;
```

Comment:
```suggestion
	if (!nmemb || !size) {
		free(ptr);
		return nullptr;
	}
```

Reply:
not sure it s appropriate to free here to be honest.

Reply:
Well, then you'd leak the user-specified `ptr`, wouldn't you?

Reply:
it is up to the caller to take care of it I think.

Reply:
Just as an example the openbsd implementation does not do it https://github.com/aseprite/giflib/blob/master/lib/openbsd-reallocarray.c

Reply:
Well, then their implementation is either very unintuitive to use (i.e. doesn't match what `realloc` does - which this is apparently modelled on) or has a bug.

Reply:
Also, here's what OpenBSD does: https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/lib/libc/stdlib/reallocarray.c, i.e. they pass the user-specified ptr to realloc which then as per 7.20.3.1 of the C11 standard frees it.

Reply:
well if your PR about realloc is accepted what would be, is to check only overflow and since the whole size is checked in realloc and frees the original. what do you think ?

Reply:
Sure, that would make sense. Which leaves us (or well, you) with @awesomekling's comment. :)